lib/evaluators/evaluatorHelpers/BaseExpressionEvaluator.ts

```diff
@@ -1,32 +1,35 @@
 import type * as RDF from '@rdfjs/types';
+import type * as LRUCache from 'lru-cache';
 import type * as E from '../../expressions';
 import { expressionToVar } from '../../functions/Helpers';
 import type { OverLoadCache } from '../../functions/OverloadTree';
 import type { ITermTransformer } from '../../transformers/TermTransformer';
 import * as Err from '../../util/Errors';
-import type { SuperTypeCallback, TypeCache, ISuperTypeProvider } from '../../util/TypeHandling';
+import type { ISuperTypeProvider, SuperTypeCallback, TypeCache } from '../../util/TypeHandling';
 
 export interface ISharedContext {
   now?: Date;
   baseIRI?: string;
-  overloadCache?: OverLoadCache;
+  /**
+   * @deprecated This value is unused, will be removed in the next major version.
+   */
+  overloadCache?: LRUCache<string, any>;
   typeCache?: TypeCache;
   getSuperType?: SuperTypeCallback;
   /**
-   * This feature is opt in. It activates the use of a new 'experimental' type system.
-   * This system is needed when using typeCache, overloadCache or getSuperType.
-   * The system is more powerful and reliable. In most cases however the old system works perfectly.
-   * Using this experimental system makes sparqlee a bit slower but more reliable using type promotion for example.
+   * @deprecated This value is unused (extended XSD types is always enabled), will be removed in the next major version.
    */
   enableExtendedXsdTypes?: boolean;
+  // This name is chosen because overloadCache was already taken by previous version.
```

Comment:
The downside of this is that we require downstream users to modify things twice, which is not so nice.
As alternative, what about just renaming `OverLoadCache` to `FunctionArgumentsCache`?

Reply:
That is better indeed! Otherwise, things might get confusing :)